### PR TITLE
engineering: concurrency in javascript

### PR DESCRIPTION
**Pull request by hieuphq**
#### What's this PR do?

- [x] concurrency in js

### CODE REVIEW

File: Engineering/Concurrency in Javascript.md [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+printSquare(4);
+```
+
+When running the above program, the Javascript will run the function step-by-step and add the function call to the `stack`. The first is: that the `printSquare()` function is added to the stack. The next functions are `square()`, `mult()`. The items in the `stack` will be executed orderly.
+
+## Javascript is a single-threaded language
+
+In reality, we always execute many tasks for a long time. These are opening a big file, downloading an image, or invoking a third-party service... We need to wait for the function call to be completed. How do we deal with the function call with the execution time too long? The Javascript runtime that is supported from the web browser, node.js. The web browser, or JS engines,... provides the APIs that help us run the task outside the `stack`. It means the long-time functions have already been finished in the stack, and the following function in the `stack` won't be blocked.
+
+```mermaid
```
Comment (zlatanpham) [inline Engineering/Concurrency in Javascript.md:40]: Not sure if it works on Obsidian

**Review by zlatanpham — APPROVED** [review]
LGTM. @monotykamary do you want to take a look?

File: Engineering/Concurrency in Javascript.md (hunk 1 above)
Comment (monotykamary) [reply Engineering/Concurrency in Javascript.md:40]: image
All good 👌

**Review by monotykamary — APPROVED** [review]
LGTM

**Pull request merged**